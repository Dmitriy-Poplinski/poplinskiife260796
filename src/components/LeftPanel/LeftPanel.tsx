import React from 'react';

import classes from './LeftPanel.module.css';

interface LeftPanelProps {
    image: string
}

export function LeftPanel({ image }: LeftPanelProps) {
    let color;

    try {
        color = `url(${require(`../../assets/images/${image}.jpg`)})`; 
    } catch (e) {
        color = 'orange'
    };

    return (
        <div 
            className={classes.left_panel}
            style={{
                background: color
            }}
            data-testid="left_panel"
        >
            Left Panel
        </div>
    );
}
