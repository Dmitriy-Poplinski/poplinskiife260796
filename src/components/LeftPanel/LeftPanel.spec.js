import { render, screen } from '@testing-library/react';

import { LeftPanel } from './LeftPanel';

describe('LeftPanel:', () => {

    let data; 

    test('LeftPanel gets incorrect string', () => {
        data = 'testing_image';

        render(<LeftPanel image={data} />);

        expect(screen.getByTestId('left_panel')).toHaveStyle('background: orange');
        expect(screen.getByTestId('left_panel')).toMatchSnapshot();
    });

    test('LeftPanel gets correct string', () => {
        data = 'left_panel_cover';

        render(<LeftPanel image={data} />);

        expect(screen.getByTestId('left_panel')).toHaveStyle(`background: url("left_panel_cover.jpg")`);
        expect(screen.getByTestId('left_panel')).toMatchSnapshot();
    });
});
