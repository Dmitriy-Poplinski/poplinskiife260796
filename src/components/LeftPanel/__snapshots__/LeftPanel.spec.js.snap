// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftPanel: LeftPanel gets correct string 1`] = `
<div
  class="left_panel"
  data-testid="left_panel"
  style="background: url(left_panel_cover.jpg);"
>
  Left Panel
</div>
`;

exports[`LeftPanel: LeftPanel gets incorrect string 1`] = `
<div
  class="left_panel"
  data-testid="left_panel"
  style="background: orange;"
>
  Left Panel
</div>
`;
